### PR TITLE
[Fix] Mixi macropad encoder orientation

### PR DESCRIPTION
**Pull request by sendz**
## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

On the Mixi macropad, the secondary encoder's orientation is wrong, it seems the pin configuration swapped, the expected behavior when rotated clockwise should be increase the value, however it will decrease the registered value.

### Expected:
* Rotate clockwise should increase the value like `Brightness Up`

### Actual:
* Rotate clockwise is decrease the value, eg: `Brightness Down`

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by ohchiko** [comment]
After looking at the schematic, the secondary encoder's pad definitions were fine. Instead, the first encoder's pad definitions were incorrect.
And also, I realized that the term _clockwise_ at QMK docs is referring to a negative value, e.g. `CW = KC_DOWN` and `CCW = KC_UP`. While in Mixi macropad is the opposite.

Wrap up, the first encoder (index 0)'s pad definitions were incorrect by the schematic. The term clockwise in QMK docs is referring to a negative value, while in this macropad is referring to a positive value. Not sure which way to proceed.

**Comment by sendz** [comment]
@ohchiko I got it, I checked the log here [2020 Feb 29 Changelog - QMK Firmware](https://docs.qmk.fm/#/ChangeLog/20200229?id=encoder-flip) and read about the issue, so the solution from me is to fix the `index 0` potentiometer and use `#define ENCODER_DIRECTION_FLIP` to flip the clockwise direction (reference: [Encoders - QMK Firmware](https://docs.qmk.fm/#/feature_encoders?id=encoders)

**Comment by ohchiko** [comment]
Tested and confirmed, the latest commit should fix the issue. Thank you.

**Comment by drashna** [comment]
In theory, this change... shouldn't do anything.  you swapped the pins *and* added the flip define. which should accomplish the same thing.

File: keyboards/mixi/config.h [hunk 1]
```diff
@@ -21,8 +21,9 @@
     { C6, F7, B6 } \
 }
 
-#define ENCODERS_PAD_A { D7, B1 }
-#define ENCODERS_PAD_B { E6, B3 }
+#define ENCODER_DIRECTION_FLIP
+#define ENCODERS_PAD_A { E6, B1 }
+#define ENCODERS_PAD_B { D7, B3 }
```
[the remark this answers is not in the document]
Comment (sendz) [reply keyboards/mixi/config.h:26]: Already propose that option, but as quoted reply from @ohchiko as the maker, I  do different approach.

> 
> 
> After looking at the schematic, the secondary encoder's pad definitions were fine. Instead, the first encoder's pad definitions were incorrect.
> And also, I realized that the term _clockwise_ at QMK docs is referring to a negative value, e.g. `CW = KC_DOWN` and `CCW = KC_UP`. While in Mixi macropad is the opposite.
> 
> Wrap up, the first encoder (index 0)'s pad definitions were incorrect by the schematic. The term _clockwise_ in QMK docs is referring to a negative value, while in this macropad is referring to a positive value. Not sure which way to proceed.
> 
> Edit: the secondary encoder (index 1)'s pad definitions were fine by the schematic.

[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/mixi/config.h:26]: That doesn't really answer anything. The pad definitions are irrelevant.
In its current state, if you need to flip the encoder direction at the keymap level you'd have to `#undef ENCODER_DIRECTION_FLIP`, which could be unintuitive vs. `#define ENCODER_DIRECTION_FLIP`.

[the remark this answers is not in the document]
Comment (ohchiko) [reply keyboards/mixi/config.h:26]: Above suggested solution will also fix the problem (please refer to my last comment). The main goal is to have both encoders same rotating direction.
If I refer to the PCB schematic, latest change of this PR for the pad definitions is correct and do the work. While in the other way removing `ENCODER_DIRECTION_FLIP` and swapped the pad definitions will also reach the main goal.

I think what's left is to either have `ENCODER_DIRECTION_FLIP` or not. I've tested to compile both changes, and removing `ENCODER_DIRECTION_FLIP` doesn't do much on the firmware size (if this suggestion is targeting a smaller firmware size). Both will have 25990/28672  (90%, 2682 bytes free) firmware size. So, for me having a correct pad definitions by the PCB schematic and define `ENCODER_DIRECTION_FLIP` will be a better solution.

**Pull request closed without merging**